lib/modules/manager/helmv3/common.ts

```diff
@@ -0,0 +1,34 @@
+import { quote } from 'shlex';
+import upath from 'upath';
+
+import type { HostRule } from '../../../types';
+import type { ExtraEnv } from '../../../util/exec/types';
+import {
+  privateCacheDir,
+} from '../../../util/fs';
+
+
+export interface OCIRepositoryRule {
+  host: string;
+  hostRule: HostRule;
+}
+
+export function getRegistryLoginCmd(
+  ociRepositoryRule: OCIRepositoryRule
+): string {
+  const { username, password } = ociRepositoryRule.hostRule;
+  if (username && password) {
+    return `helm registry login --username ${quote(username)} --password ${quote(password)} ${ociRepositoryRule.host}`
+  }
+  return ""
+}
```

Comment:
```suggestion
export function generateLoginCmd(
  repositoryRule: RepositoryRule,
  loginCMD: string
): string | null {
  const { username, password } = RepositoryRule.hostRule;
  if (username && password) {
    return `${loginCMD} --username ${quote(username)} --password ${quote(password)} ${repositoryRule.host}`
  }
  return null
}
```
- `get` suggest a variable retrieval
- use RepositoryRule rather than creating a own type
- do not return empty strings and rather return null 
- parameterize the command so we can use it for registries and repositories.

You can wrap this `generateRegistryLoginCMD` and `generateRepositoryLoginCMD` for usage in `helmfile`